### PR TITLE
Filtering subscriptions and iap products

### PR DESCRIPTION
**Pull request by andresesfm**
Fixes https://github.com/dooboolab/react-native-iap/issues/1697. Only including subscriptions/iap products in the result array

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -40,6 +40,8 @@ export enum InstallSourceAndroid {
 }
 
 export interface ProductCommon {
+  type: 'subs' | 'sub' | 'inapp' | 'iap';
```
Comment (hyochan) [inline src/types/index.ts:43]: We currently have 'iap' and 'sub' for iOS and
`inapp` and `subs' for [android which is the official type](https://developer.android.com/reference/com/android/billingclient/api/BillingClient.SkuType).

Looks like we can unify this.

Looks

**Review by hyochan — APPROVED** [review]
LGTM!

**Comment by Dallas62** [comment]
This PR just beak subscriptions, but who cares...

image

Stop try to ONE-LINE everythings, you just filtered a Promise which leads to `undefined undefined is not a function [TypeError: undefined is not a function]`

**Comment by hyochan** [comment]
> This PR just beak subscriptions, but who cares...
> 
> image
> 
> Stop try to ONE-LINE everythings, you just filtered a Promise which leads to `undefined undefined is not a function [TypeError: undefined is not a function]`

Oh my fault 😱  I should have review more carefully.

So the code should be
```
(await getIosModule()
          .getItems(skus))
          .filter((item: ProductCommon) => skus.includes(item.productId))
          .filter((item: ProductCommon) => item.type === 'iap');
```

@Dallas62 Could you kindly check if this works?

**Comment by Dallas62** [comment]
Make it more readable, to avoid confusion:

```js
const items = await getIosModule().getItems(skus);

return items.filter((item: ProductCommon) => skus.includes(item.productId))
          .filter((item: ProductCommon) => item.type === 'iap');
```

(format/indent like you want)

**Comment by hyochan** [comment]
> Make it more readable, to avoid confusion:
> 
> ```js
> const items = await getIosModule().getItems(skus);
> 
> return items.filter((item: ProductCommon) => skus.includes(item.productId))
>           .filter((item: ProductCommon) => item.type === 'iap');
> ```
> 
> (format/indent like you want)

Would you kindly submit a PR for this?

**Pull request merged**